File: backend/src/main/java/com/propcoza/legends/tools/rental_transaction_manager/dto/InstanceReturnDto.java
package com.propcoza.legends.tools.rental_transaction_manager.dto;

import com.propcoza.legends.tools.rental_transaction_manager.entity.InstanceStatus;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@Setter
public class InstanceReturnDto {

    private UUID id;
    private UUID rentalId;
    private LocalDate billingPeriod;
    private LocalDate actualPaymentDate;

    // -------------------------------
    //     Commission Percentages (Changed to BigDecimal to avoid unboxing crashes)
    // -------------------------------
    private BigDecimal rentalCommissionPercent;
    private BigDecimal officeSplit;
    private BigDecimal agentSplit;
    private BigDecimal agentPaye;

    // -------------------------------
    //     Financial Snapshot
    // -------------------------------
    private BigDecimal baseRent;
    private BigDecimal totalAmountPaid;
    private BigDecimal landlordPayAmount;
    private BigDecimal baseComm;
    private BigDecimal vat;
    private BigDecimal commExclVat;
    private BigDecimal companyComm;
    private BigDecimal agentGrossComm;
    private BigDecimal payeAmount;
    private BigDecimal agentNettComm;

    // -------------------------------
    //     Lease Fee
    // -------------------------------
    private BigDecimal leaseFee;
    private BigDecimal leaseFeeAgentPortion;
    private BigDecimal leaseFeeOfficePortion;

    // -------------------------------
    //     Optional Manual Entries
    // -------------------------------
    private BigDecimal deposit;

    // -------------------------------
    //     Instance Metadata (Type-safe Enum or String, without JPA annotation)
    // -------------------------------
    private InstanceStatus status;

    private String createdBy; // Fixed naming convention capitalization
    private LocalDateTime createdAt;
    private String lastModifiedBy;
    private LocalDateTime lastModifiedAt;

    // Consider adding these if the frontend needs adjustment/note info on return
    // private List<AdjustmentReturnDto> adjustments;
    // private List<NoteReturnDto> notes;
}

File: frontend/src/features/instances/InstanceReturnDto.ts
export type InstanceStatus = "APPROVED" | "CANCELLED" | "DRAFT";


export interface InstanceReturnDto {
  id: string;
  rentalId: string;
  billingPeriod: string;
  actualPaymentDate: string;

  // -------------------------------
  //     Commission Percentages
  // -------------------------------

  rentalCommissionPercent: number;
  officeSplit: number;
  agentSplit: number;
  agentPaye: number;
  vatRegistered: boolean;

  // -------------------------------
  //     Financial Snapshot
  // -------------------------------

  baseRent: number;
  totalAmountPaid: number;
  landlordPayAmount: number;
  baseComm: number;
  vat: number;
  commExclVat: number;
  companyComm: number;
  agentGrossComm: number;
  payeAmount: number;
  agentNettComm: number;

  // -------------------------------
  //     Lease Fee
  // -------------------------------

  leaseFee: number;
  leaseFeeAgentPortion: number;
  leaseFeeOfficePortion: number;

  // -------------------------------
  //     Optional Manual Entries
  // -------------------------------

  deposit: number;

  // -------------------------------
  //     Instance Metadata
  // -------------------------------

  status: string;
  createdBy: string;
  createdAt: string; // LocalDateTime (ISO format)
  lastModifiedBy: string; // LocalDateTime (ISO format)
  lastModifiedAt: string; // LocalDateTime (ISO format)
}